### ISSUE TITLE
Tasks using Docker cannot use files on local data providers

### ISSUE THREAD

**Issue by glatard** [aces/cbrain#542]
Task using Docker currently mount the following directories through `-v` options in the `docker run` command:
* present working directory (a.k.a task execution directory).
* bourreau dp cache, so that synchronized files are accessible from the container.
* task directory, so that symbolic link "DP_Cache" (that points to the bourreau dp cache) can be resolved in the container.

However, in case of a local data provider, the files in the execution directory are symlinked directly to the data provider directory which is none of the above. Thus, the symlnks to these files are broken. 

To fix this issue, one should:
* Test if the task input files are located on a local data provider.
* If yes, add the base path of this data provider to the list of directories that will be mounted in the container.

**Comment by prioux** [comment]
The data provider method to invoke is `is_fast_syncing?`

```ruby
   userfile.data_provider.is_fast_syncing?
```

`true` means the DP is local.

The method will have to be invoked for each an every input file of the current task (because a task can work with an arbitrary list of userfiles, and each of them can be on a different data provider); however to avoid repeating the call N times for the same DP, we should collect first the list of data providers used by the files (without repetition) . Something like

```ruby
dps = DataProvider.find(inputfiles.map(&:data_provider_id).uniq)
```

**Comment by glatard** [comment]
If that's possible, we might as well mount all the local data providers accessible from the current Bourreau. Mounting has no real overhead and it will make things simpler and easier to debug.

**Comment by prioux** [comment]
Yes, you're right, might be a much easier solution.

**Comment by prioux** [comment]
```ruby
dirs = DataProvider.where(:online => true).select { |dp| dp.is_fast_syncing? }.map(&:remote_dir).select { |dir| File.directory?(dir) }
```

**Comment by shots47s** [comment]
@glatard, is this issue blocking any of your work, trying to decide prioritization of this task.

**Comment by natacha-beck** [comment]
@prioux  does this PR https://github.com/aces/cbrain/pull/707 close this issue.

**Comment by prioux** [comment]
Fixed by #707
